### ISSUE TITLE
Error when quitting to menu

### ISSUE THREAD

**Issue by Danatr0nika** [OpenRCT2/OpenRCT2#17938]
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

OpenRCT2, v0.4.1-233-g306cc65 (306cc65 on develop) provided by GitHub

### Describe the issue

Quitting a game to the menu causes an error:
"An assertion failed, please report this to the OpenRCT2 developers. 
Version: OpenRCT2, v0.4.1-233-g306cc65 (306cc65 on develop) 
Location: RegisterLoadedObject:282"

Retrying can sometimes cause a successful recovery to the main menu, but other times it just crashes the game.

### Steps to reproduce

Load a park file (one is attached where I know I encountered the bug) and press "Quit to menu" in the settings dropdown until the error pops up (it may take a few tries).

### Attachments

![image](https://user-images.githubusercontent.com/43836358/187651140-e5147571-be80-4cb6-b0f1-35b079aa532d.png)
[Diamond Heights Win.zip](https://github.com/OpenRCT2/OpenRCT2/files/9460477/Diamond.Heights.Win.zip)

**Comment by Danatr0nika** [comment]
No, I've had it happen even when I load directly into the park and then immediately quit. In case any of this helps narrow it down, I'm using the Steam version of the RCT2 files and I'm on an Asus Zephyrus G14 with a Ryzen cpu, a 2060, and 16gb ram.

**Comment by Danatr0nika** [comment]
Sorry, hit the wrong button. Stupid phone.

**Comment by GitRDonne** [comment]
I'm experiencing the same issue, though I've narrowed it down somewhat to occuring only when I start a game with a classic scenario (DLC) from the first game. The RCT2 scenarios appear to quit to the menu properly.

**Comment by jlw4049** [comment]
This happens to me every time I quit to menu when hosting a server

**Comment by Jordanjtt97** [comment]
I’m having the same issue.  I click “quit to menu” and it freezes in full screen leaving me unable to do anything but restart my pc to fix the issue.  I get the error message but can’t even access it to report the failure because of the full screen game frozen.

**Comment by Gymnasiast** [comment]
Are you still having this issue?
